django_freeradius/api/views.py

```diff
@@ -186,6 +189,20 @@ def create(self, request, *args, **kwargs):
         else:
             raise ValidationError(serializer.errors)
 
+    def perform_create(self, serializer):
```

Comment:
ok we can leave this here but you know I'm picky, please rewrite this block as follows:

```python
if app_settings.API_ACCOUNTING_AUTO_GROUP:
    def perform_create(self, serializer):
        user_model = get_user_model()
        username = serializer.validated_data.get('username', '')
        try:
            user = user_model.objects.get(username=username)
        except User.DoesNotExist:
            logging.info('no corresponding user found '
                         'for username: {}'.format(username)
            serializer.save()
        else:
            group = user.radiususergroup_set.order_by('priority').first()
            # user may not have a group defined
            groupname = group.groupname if group else None
            serializer.save(groupname=groupname)
```

Reply:
I'll do this, but I was curious why in this particular way?

Reply:
I tried this, but the test I wrote which is supposed to check the behavior if the  setting is disabled failed i.e. the function is filling the group name even if for the test I set the setting to False. I'll investigate further.

Reply:
I see, it's because the `if` block is executed only when the app is booted, so even if you turn off the value of the setting during tests, the method will still be defined and will work anyway, I'll be more flexible and propose this other approach which should work in tests too:

```python
def perform_create(self, serializer):
    if app_settings.API_ACCOUNTING_AUTO_GROUP:
        user_model = get_user_model()
        username = serializer.validated_data.get('username', '')
        try:
            user = user_model.objects.get(username=username)
        except User.DoesNotExist:
            logging.info('no corresponding user found '
                         'for username: {}'.format(username)
            serializer.save()
        else:
            group = user.radiususergroup_set.order_by('priority').first()
            # user may not have a group defined
            groupname = group.groupname if group else None
            serializer.save(groupname=groupname)
    else:
        return super().perform_create(serializer)
```

Why this approach? basically, if the setting is turned off, we avoid altering the logic of this method entirely. It introduces less risks of unforeseen bugs.

Reply:
Understood @nemesisdesign , I made the changes.